godel_process_planning/src/keyence_process_planning.cpp

```diff
@@ -0,0 +1,152 @@
+#include <godel_process_planning/godel_process_planning.h>
+
+#include <ros/console.h>
+
+// descartes
+#include "descartes_trajectory/cart_trajectory_pt.h"
+#include "descartes_planner/dense_planner.h"
+
+#include "common_utils.h"
+
+// FILE LOCAL CONSTANTS
+const static std::string JOINT_TOPIC_NAME = "joint_states"; // ROS topic to subscribe to for robot state
+
+namespace godel_process_planning
+{
+
+/**
+ * @brief Translated an Eigen pose to a Descartes trajectory point appropriate for the scan process!
+ *        Mirros the function in blend_process_planning.cpp document.
+ * @param pose
+ * @param dt The upper limit of time from the previous point to achieve this one
+ * @return A descartes trajectory point encapsulating a move to this pose
+ */
+static inline descartes_core::TrajectoryPtPtr toDescartesPt(const Eigen::Affine3d& pose, double dt)
+{
+  using namespace descartes_trajectory;
+  using namespace descartes_core;
+  const descartes_core::TimingConstraint tm (dt);
+
+  Eigen::Vector3d rpy = pose.rotation().eulerAngles(0,1,2);
+  Eigen::Vector3d xyz = pose.translation();
+
+  double rx = rpy(0), ry = rpy(1), rz = rpy(2);
+  double x = xyz(0), y = xyz(1), z = xyz(2);
+
+  // By specifying a range of 180 degrees and a step of 180 degrees, the system will sample
+  // at the nominal pose +/- 90 degrees for scanning purposes.
+  TolerancedFrame frame (pose,
+                         ToleranceBase::zeroTolerance<PositionTolerance>(x,y,z),
+                         ToleranceBase::createSymmetric<OrientationTolerance>(rx,ry,rz,0,0,M_PI));
+
+  return TrajectoryPtPtr(new CartTrajectoryPt(frame, 0.0, M_PI, tm));
+}
+
+/**
+ * @brief transforms an input, in the form of a reference pose and points relative to that pose, into Descartes'
+ *        native format. Also adds in associated parameters.
+ * @param ref The reference posed that all points are multiplied by. Should be in the world space of the keyence move group.
+ * @param points Sequence of points (relative to ref and the world space of blending robot model)
+ * @param params Surface blending parameters, including info such as traversal speed
+ * @return The input trajectory encoded in Descartes points
+ */
+static godel_process_planning::DescartesTraj toDescartesTraj(const geometry_msgs::Pose& ref,
+                                                             const std::vector<geometry_msgs::Point>& points,
+                                                             const godel_msgs::ScanPlanParameters& params)
+{
+  DescartesTraj traj;
+  traj.reserve(points.size());
+  if (points.empty()) return traj;
```

Comment:
Not here, but you make me realize that the main planning methods did not check for zero-sized inputs. I've updated them to add these checks and those do warn and return false.